src/net/connector.rs

```diff
@@ -33,11 +41,20 @@ impl UdpConnector {
 
 impl NetworkConnector for UdpConnector {
     type Stream = UdpSender;
-    
+
     fn connect(&self, host: &str, port: u16, _: &str) -> error::Result<<Self as NetworkConnector>::Stream> {
         let udp_sock = try!(self.0.try_clone());
-        let sock_addr = try!(net::addr_from_trait((host, port)));
-        
+        let sock_addr = match try!(self.local_addr()) {
+            SocketAddr::V4(_) => SocketAddr::V4(SocketAddrV4::new(FromStr::from_str(host).unwrap(), port)),
+            SocketAddr::V6(n) => {
+                let mut addr: SocketAddrV6 = FromStr::from_str(format!("{}:{}", host, port).as_str())
```

Comment:
I think the current state of it is slightly safer than calling connect with what should be a _bad_ ip address. Either case works for now, but if hyper fixes their "[host]:port" split code then the connect would break if we had it hardcoded to expect the "[]" as part of host.

> Btw, we should file a bug on hyper...